paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentSheetViewModelTest.kt

```diff
@@ -2766,6 +2768,34 @@ internal class PaymentSheetViewModelTest {
             val screen = awaitItem()
             assertThat(screen).isInstanceOf<PaymentSheetScreen.CvcRecollection>()
         }
+        FakeCvcRecollectionInteractor.subscriptionCount().test {
+            assertThat(awaitItem()).isEqualTo(1)
+        }
+    }
+
+    @Test
+    fun `CvcRecollection state update should update payment selection`() = runTest {
+        val viewModel = createViewModel(
+            args = ARGS_CUSTOMER_WITH_GOOGLEPAY.copy(
+                config = ARGS_CUSTOMER_WITH_GOOGLEPAY.config.copy(
+                    paymentMethodLayout = PaymentSheet.PaymentMethodLayout.Vertical
+                )
+            )
+        )
+
+        cvcRecollectionHandler.requiresCVCRecollection = true
+        cvcRecollectionHandler.cvcRecollectionEnabled = true
+        viewModel.checkout()
+
+        viewModel.selection.test {
+            awaitItem()
+
+            FakeCvcRecollectionInteractor.updateCompletionState(CvcCompletionState.Completed("444"))
```

Comment:
We don't want our tests to mutate global state. Please make this an instance variable.